### PR TITLE
Feature/1200453899422605/expand date filters

### PR DESCRIPTION
**Pull request by jrrlokken**
## Description

Added additional DatePicker components and logic to append all required dates to db query, allowing for filtering organizations by date ranges for last_verified, last_updated and created_at dates.

## PR Checklist

- [x] Assign @FJKhan **and** @Alfredo-Moreira as reviewers.
- [x] If your PR is not a hotfix, is it targeted for `dev`? If your PR is a hotfix, is it targeted for `master`?
- [ ] Unit and functional test coverage was added where applicable.
- [x] CI/CD passes for your PR.
- [ ] Complex code is well documented with comments.
- [x] Does the original ticket have test instructions? If not add them below

## How to Test

1. Login to Control Panel.
2. Under Filter Organizations, enter a date or use the date picker to populate either the first date field or both date fields under Last Verified, Last Updated or Created At
3. Click Search
4. Click View next to one of the returned results to view the organization
5. If only the first date field was used, verify that the returned results have a [Last Verified, Last Updated or Created At] dates older than the date entered. If both date fields for a criteria were used, verify that the returned results have a [Last Verified, Last Updated or Created At] date between the date values entered.

### CODE REVIEW

File: src/date-picker.css [hunk 1]
```diff
@@ -9,10 +9,12 @@
 }
 
 .react-datepicker__input-container {
-  font-size: 1rem;
-  padding-left: 1rem;
-  padding-right: 1rem;
+  font-size: 0.8rem;
+  padding-left: 0.5rem;
+  padding-right: 0.5rem;
+  /* margin: 0 0 0 2rem; */
```
Comment (FJKhan) [inline src/date-picker.css:15]: can you remove the lines instead of commenting them out, please?

**Review by FJKhan — CHANGES_REQUESTED** [review]

File: src/date-picker.css (hunk 1 above)
Comment (jrrlokken) [reply src/date-picker.css]: You bet -- done.

File: src/components/FiltersOrganizations.js [hunk 2]
```diff
@@ -185,11 +207,23 @@ const FiltersOrganizations = (props) => {
...
+        <DateFieldPicker
+          selected={lastUpdatedEnd}
+          onChange={setLastUpdatedEnd}
+        />
+
+        <Text>Created At:</Text>
+        <DateFieldPicker selected={createdAt} onChange={setCreatedAt} />
+        <DateFieldPicker selected={createdAtEnd} onChange={setCreatedAtEnd} />
 
         <Text>Created Before:</Text>
```
Comment (FJKhan) [inline src/components/FiltersOrganizations.js:228]: remember to remove this previous filter.

**Review by FJKhan — CHANGES_REQUESTED** [review]
can we get better labels on the filters please? I can't tell which is for "before" and "after".
also, there are now 3 filters related to the "Created at" date. two of which are I'm assuming for "created before".

image

File: src/components/FiltersOrganizations.js (hunk 2 above)
Comment (jrrlokken) [reply src/components/FiltersOrganizations.js]: Extraneous date filter removed.

**Comment by jrrlokken** [comment]
Improved labeling/layout of date filters:

**Review by FJKhan — CHANGES_REQUESTED** [review]
@jrrlokken thanks for the updates! this is looking so much better! Amazing job, bud!

a few things I noticed:
 - the placeholder text for using a single date vs date range are different. I think both should say "select x date"
 
image

- when I switch from using a date range to a single date, the second (end) date value is retained which leads to incorrect search results
 
image

**Comment by jrrlokken** [comment]
Addressed concerns about placeholder text, it is much simpler now, single date picker shows 'Select a date', date range pickers show 'Select start date' and 'Select end date'.  Validations have been implemented to prevent leaving behind stale dates and to prevent invalid or out-of-scope dates.

**Review by FJKhan — APPROVED** [review]

**Pull request merged**